### PR TITLE
Cube Overview Bugfixes

### LINKED ISSUE #854: Capitals in custom id cause links to break
We need to use a .tolowercase() on the edit overview submit route in order to accommodate this

### LINKED ISSUE #861: Overview Modal doesn't load all fields
Overview Modal does not load the description correctly if the description is not using html.
Overview Modal also does not load the custom url alias

### PR DESCRIPTION
**Pull request by dekkerglen**
Fixes #861, Fixes #854

### CODE REVIEW

File: src/components/CubeOverviewModal.js [hunk 1]
```diff
@@ -26,14 +26,17 @@ class CubeOverviewModal extends Component {
   constructor(props) {
     super(props);
 
+    props.cube.descriptionhtml= props.cube.descriptionhtml && props.cube.descriptionhtml !== 'undefined' ? props.cube.descriptionhtml : props.cube.description;
```
Comment (phulin) [inline src/components/CubeOverviewModal.js]: This totally breaks React's data model and you should not do this (meaning modifying incoming props).

File: src/components/CubeOverviewModal.js [hunk 2]
```diff
@@ -368,7 +368,7 @@ class CubeOverviewModal extends Component {
                 <h6>Description</h6>
                 <TextEntry
                   name="blog"
-                  value={this.state.cube.descriptionhtml ? this.state.cube.descriptionhtml : ''}
+                  value={cube.descriptionhtml && cube.descriptionhtml !== 'undefined' ? cube.descriptionhtml :cube.description}
```
Comment (phulin) [inline src/components/CubeOverviewModal.js]:
```suggestion
                  value={typeof cube.descriptionhtml !== 'undefined' ? cube.descriptionhtml :cube.description}
```

Comment (dekkerglen) [reply src/components/CubeOverviewModal.js]: What is this hack

Comment (phulin) [reply src/components/CubeOverviewModal.js]: This is actually how you are supposed to check if something is undefined

Comment (dekkerglen) [reply src/components/CubeOverviewModal.js]: The issue is descriptionhtml can be set to the string value of 'undefined' if any other field was updated with the new modal

Comment (dekkerglen) [reply src/components/CubeOverviewModal.js]: So this checks if it's actually undefined, or if it is equal to the string value of undefined

**Comment by phulin** [comment]
Oh wat i don’t like that pattern but this pr is fine with me then

**Comment by dekkerglen** [comment]
I need to beautify it, will do that once I get home

**Pull request merged**